src/models/data.py

```diff
@@ -42,6 +43,16 @@ def concatenate(self, x: TrainData):
                     newval = torch.cat((val, getattr(x, key)), dim=0)
                 setattr(self, key, newval)
 
+    def filter(self, filter_array: Union[np.ndarray, torch.Tensor]) -> None:
```

Comment:
What is this function doing?

Reply:
Before, we had a bunch of lines which were
`historical = historical[notnan_indices]`

now, we can filter the entire `TrainData` object, so

`traindata.filter(notnan_indices)`

Reply:
gotcha